### PR TITLE
Remove the execute permission from the directory

### PR DESCRIPTION
**Pull request by zishen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Reduce permissions for directories created by karmada itself.
From 755 to 700. Prevent unauthorized users from modifying content.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

@RainbowMango @whitewindmills @XiShanYongYe-Chang

### CODE REVIEW

File: pkg/karmadactl/cmdinit/kubernetes/deploy.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func initializeDirectory(path string) error {
 			return err
 		}
 	}
-	if err := os.MkdirAll(path, os.FileMode(0o755)); err != nil {
+	if err := os.MkdirAll(path, os.FileMode(0o640)); err != nil {
```
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/kubernetes/deploy.go:222]:
```suggestion
	if err := os.MkdirAll(path, os.FileMode(0640)); err != nil {
```

**Comment by RainbowMango** [comment]
CI is failing:
```
I0824 09:28:14.555201   18831 deploy.go:184] kubeconfig file: /home/runner/.kube/karmada-host.config, kubernetes: https://172.18.0.5:6443/
I0824 09:28:14.63[56](https://github.com/karmada-io/karmada/actions/runs/5961906734/job/16172031967?pr=3992#step:5:57)89   18831 deploy.go:204] karmada apiserver ip: [172.18.0.5]
error: certificate generation failed.unable to write private key to file mkdir /home/runner/karmada/pki: permission denied
```

**Comment by zishen** [comment]
> CI is failing: https://github.com/karmada-io/karmada/actions/runs/5961906734/job/16172031967?pr=3992
> 
> ```
> I0824 09:28:14.555201   18831 deploy.go:184] kubeconfig file: /home/runner/.kube/karmada-host.config, kubernetes: https://172.18.0.5:6443/
> I0824 09:28:14.63[56](https://github.com/karmada-io/karmada/actions/runs/5961906734/job/16172031967?pr=3992#step:5:57)89   18831 deploy.go:204] karmada apiserver ip: [172.18.0.5]
> error: certificate generation failed.unable to write private key to file mkdir /home/runner/karmada/pki: permission denied
> ```

yea, Directory requires execute permission, otherwise files cannot be created.
so, I changed the permission to 700.

**Comment by chaosi-zju** [comment]
```console
runner@fv-az454-559:~$ pwd
/home/runner
runner@fv-az454-559:~$ mkdir /home/runner/karmada
runner@fv-az454-559:~$ chmod -R 640 /home/runner/karmada
runner@fv-az454-559:~$ mkdir /home/runner/karmada/pki
mkdir: cannot create directory ‘/home/runner/karmada/pki’: Permission denied
```

As you see, if you chmod `640` to one directory, the directory equal to readonly, the mod of directory should be at lease `700`~

more example as follows:

```console
runner@fv-az454-559:~$ mkdir /home/runner/karmada
runner@fv-az454-559:~$ chmod -R 700 /home/runner/karmada
runner@fv-az454-559:~$ mkdir /home/runner/karmada/pki
runner@fv-az454-559:~$ chmod -R 700 /home/runner/karmada/pki
runner@fv-az454-559:~$ touch /home/runner/karmada/pki/xx.txt 
runner@fv-az454-559:~$ ls /home/runner/karmada/pki/xx.txt
/home/runner/karmada/pki/xx.txt
```

```console
runner@fv-az454-559:~$ mkdir /home/runner/karmada
runner@fv-az454-559:~$ chmod -R 700 /home/runner/karmada
runner@fv-az454-559:~$ mkdir /home/runner/karmada/pki
runner@fv-az454-559:~$ chmod -R 640 /home/runner/karmada/pki
runner@fv-az454-559:~$ touch /home/runner/karmada/pki/xx.txt 
touch: cannot touch '/home/runner/karmada/pki/xx.txt': Permission denied
```

`700` is enough in our case, while, directory usually chmod to `750` in general.

**Comment by chaosi-zju** [comment]
image

I think you may not changed completed~

**Comment by zishen** [comment]
> image
> 
> I think you may not changed completed~

yea, thanks a lot.

**Review by vincentgoat — APPROVED** [review]
lgtm

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**